play-services-core/src/main/java/org/microg/gms/gcm/McsService.java

```diff
@@ -166,6 +168,26 @@ public void onCreate() {
         heartbeatIntent = PendingIntent.getService(this, 0, new Intent(ACTION_HEARTBEAT, null, this, McsService.class), 0);
         alarmManager = (AlarmManager) getSystemService(ALARM_SERVICE);
         powerManager = (PowerManager) getSystemService(POWER_SERVICE);
+        if(mDeviceIdleController == null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            try {
+                java.lang.reflect.Method method = Class.forName("android.os.ServiceManager").getMethod("getService", String.class);
+                java.lang.reflect.Field field = Context.class.getField("DEVICE_IDLE_CONTROLLER");
+                IBinder binder = (IBinder) method.invoke(null, field.get(null));
+                if(binder != null)
+                    mDeviceIdleController = IDeviceIdleController.Stub.asInterface(binder);
+            } catch (NoSuchFieldException e) {
+                e.printStackTrace();
```

Comment:
You shouldn't use `e.printStackTrace()` as it logs to stdout which shouldn't be used on Android. Use `Log.w(TAG, e)` instead.